### PR TITLE
Adds support for global variables for CNS

### PR DESCRIPTION
**Pull request by joaomcteixeira**
Adds global variables for CNS runs according to what we discussed.

@amjjbonvin can you read over the documentation of `CNSJob` and see if they agree with expectations? Also, can you test out the examples? With this changes the `rigid-body` example broke. Likely because the `RUN` variable does not point to the same place as before. Should we update the var or the CNS scripts?

**EDIT:** likely the example broke because of #132

### CODE REVIEW

**Review by rvhonorato — APPROVED** [review]

File: src/haddock/libs/libsubprocess.py [hunk 1]
```diff
@@ -45,20 +46,45 @@ def __init__(
...
             cns_exec=None,
             ):
         """
         CNS subprocess.
 
-        :param input_file: input CNS script
-        :param output_file: CNS output
-        :cns_folder: absolute execution path
-        :cns_exec: CNS binary including absolute path
+        Parameters
```
Comment (amjjbonvin) [inline src/haddock/libs/libsubprocess.py:57]: Shouldn't this also document the toppa_path?

Comment (joaomcteixeira) [reply src/haddock/libs/libsubprocess.py:57]: Okay, will commit on that.

File: src/haddock/libs/libsubprocess.py [hunk 2]
```diff
@@ -45,20 +46,45 @@ def __init__(
...
+
+        cns_folder : str of pathlib.Path
+            The path where the CNS scripts needed for the module reside.
+            For example, `modules/rigidibody/cns`.
+
+        mod_path : str of pathlib.Path
+            Path where the results of the haddock3 module executing this
+            CNS job will be saved.
+
+        config_path : str of pathlib.Path
```
Comment (amjjbonvin) [inline src/haddock/libs/libsubprocess.py:74]: Here you mean the user configuration file, right?

Comment (joaomcteixeira) [reply src/haddock/libs/libsubprocess.py:77]: Yes, the path pointing to the user configuration file. But we need to folder, not the file itself. Let me check that.

**Comment by joaomcteixeira** [comment]
Have a look to everything now. But the `rigid-body` example is not working on my side.

**Comment by joaomcteixeira** [comment]
@amjjbonvin question: the `toppar` is the one in `cns/toppar` or the one in `modules/sampling/rigidbody/cns/toppar` ?

**Comment by amjjbonvin** [comment]
That’s normal - need to correct the scripts

**Comment by amjjbonvin** [comment]
> 
> the one in cns/toppar 

I will delete the other dirs once I update the scripts

**Comment by joaomcteixeira** [comment]
Okay, now it is the `cns/topppar` as you can see `toppar` is imported from : https://github.com/haddocking/haddock3/blob/2eeab502c56e95fb543606b7bc6289ce3abf1ea9/src/haddock/__init__.py#L16

Feel free to merge this PR if you feel it does the job. So you can continue the CNS progress.

**Review by amjjbonvin — APPROVED** [review]

**Pull request merged**